### PR TITLE
Document how to add new entrainment closures

### PR DESCRIPTION
**Pull request by costachris**
Document process of adding a new entrainment/detrainment closure

### CODE REVIEW

**Comment by costachris** [comment]
@haakon-e  You can see the generated docs by clicking `Details` in the `documenter/deploy` GHA above.

File: docs/src/Closures.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+## An example: Simple Neural Network Closure
+
+Here we've added a simple neural network which maps non-dimensional groups (`nondim_groups`) to the non-dimensional components
+of entrainment & detrainment (`nondim_ε` & `nondim_δ`, respectively). 
+The entrainment type is `NNEntr` and the namelist name is "NN".
+
+
+#### src/closures/entr_detr.jl
+```
```
Comment (charleskawczynski) [inline docs/src/Closures.md:80]:
```suggestion
```julia
```

Comment (charleskawczynski) [reply docs/src/Closures.md:80]: This will give the code snippets syntax highlighting 🙂

**Comment by charleskawczynski** [comment]
This section: `Submitting TC.jl on the Caltech cluster` might be a nice section on its own

File: docs/src/Closures.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
+# Adding closures to EDMF
+
+Entrainment closure functions are written in a file that can be augmented with additional closure types. The type of entrainment closure is specified in the namelist, which is a collection of model parameters, flags, and other options used to run EDMF. After the new closure is implemented, EDMF can be run with your new closure by running `julia --project integration_tests/LES_driven_SCM.jl` in the top level of the repo (locally), or with the submit script below on the Caltech HPC cluster.
+
+## Adding a general closure function:
+*1.)* Add a `entr_detr` function defintion to `src/closures/entr_detr.jl` which implements your closure. The code uses multiple 
+dispatch to determine which type of entrainment to used based on the `εδ_model_type` argument. All `entr_detr` 
+definitions must take:
+		`param_set` (a collection of parameters, many of which are defined in the namelist, as well as parameters needed for your closure)
+		`εδ_model_vars` - a structure containing all input variables your closure might depend on.
```
Comment (yairchn) [inline docs/src/Closures.md:10]: if someone would like to learn from / use profiles of variables that are not in 'εδ_model_vars' right now they would have to add these variables to 'εδ_model_vars' which is both in 'update_aux' in where 'εδ_model_vars' type is defined, right ?

**Review by yairchn — APPROVED** [review]

File: docs/src/Closures.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    c_gen = ICP.c_gen(param_set)
+    dim_scale = dimensional_part(param_set, εδ_model_vars)
+    area_limiter = max_area_limiter(param_set, εδ_model_vars)
+
+    # Neural network closure
+    nn_arc = (4, 2, 2)  # (#inputs, #neurons, #outputs)
+    nn_model = Chain(
+        Dense(reshape(c_gen[1:8], nn_arc[2], nn_arc[1]), c_gen[9:10], sigmoid),
+        Dense(reshape(c_gen[11:14], nn_arc[3], nn_arc[2]), c_gen[15:16], softplus),
+    )
```
Comment (charleskawczynski) [inline docs/src/Closures.md:98]:
```suggestion
    nn_model = Flux.Chain(
        Flux.Dense(reshape(c_gen[1:8], nn_arc[2], nn_arc[1]), c_gen[9:10], sigmoid),
        Flux.Dense(reshape(c_gen[11:14], nn_arc[3], nn_arc[2]), c_gen[15:16], softplus),
    )
```

**Pull request closed without merging**